Spells/script/PlayerConfigStorage.js

```diff
@@ -0,0 +1,106 @@
+
+// Persists player stats in browser storage, meant to be shared between tools
+class PlayerConfigStorage
+{
+    static IDENTIFIER = "ELDEN_RING_TOOLS_PLAYER_CONFIG";
+    static INSTANCE = new PlayerConfigStorage();
+
+    _config = PlayerConfigStorage.getDefaultConfig();
+    _loaded = false;
+
+    static getInstance()
+    {
+        return PlayerConfigStorage.INSTANCE;
+    }
+
+    getConfig()
+    {
+        if (!this._loaded)
+        {
+            this._config = PlayerConfigStorage.loadConfiguration();
+            this._loaded = true;
+        }
+        return this._config;
+    }
+
+    setConfig(value)
+    {
+        this._config = value;
+        PlayerConfigStorage.saveConfiguration(value);
+    }
+
+    static loadConfiguration()
+    {        
+        const config = localStorage.getItem(PlayerConfigStorage.IDENTIFIER);
+        if (config != null)
+            return JSON.parse(config);
+        else
+            return PlayerConfigStorage.getDefaultConfig();
+    }
+
+    static saveConfiguration(config)
+    {
+        localStorage.setItem(PlayerConfigStorage.IDENTIFIER, JSON.stringify(config));
+    }
+
+    // This isn't used anywhere, it's just for reference and to provide type-checking
+    // (TypeScript automatically infers the type of the return value, which we want to apply to the "config" property)
```

Comment:
Minor note: while I eventually want to get into TypeScript (been meaning to learn it forever), I don't think this project will be able to make the transition.

I do like what I'm seeing w/jsconfig, though.

Reply:
Ah, that was to say that even without using TypeScript, IDE's like VS Code can infer the object type:

![image](https://user-images.githubusercontent.com/26604994/170767436-a2f9a284-3fef-47b2-af50-458036554524.png)